subsys/bluetooth/services/hids.c

```diff
@@ -1330,8 +1338,9 @@ int bt_hids_boot_kb_inp_rep_send(struct bt_hids *hids_obj,
 	rep_data = conn_data->hids_boot_kb_inp_rep_ctx;
 
 	memcpy(rep_data, rep, len);
-	memset(&rep_data[len], 0,
-	       (sizeof(conn_data->hids_boot_kb_inp_rep_ctx) - len));
+	if (len < sizeof(conn_data->hids_boot_kb_inp_rep_ctx)) {
+		memset(&rep_data[len], 0, (sizeof(conn_data->hids_boot_kb_inp_rep_ctx) - len));
+	}
```

Comment:
Is it necessary? It seems that we checked the following condition a few lines above:
```len > sizeof(conn_data->hids_boot_kb_inp_rep_ctx)```
Here we can have a memset call with the length parameter equal to zero, in which case it won't do anything. Or am I missing something?

Reply:
Actually we can't have len = 8, because it passes pointer out of rep_data bounds (it's length is 8, so [8] is out of bounds].